esiclient/v1/node_network.py

```diff
@@ -87,23 +87,39 @@ def get_parser(self, prog_name):
             metavar="<node>",
             help=_("Name or UUID of the node"))
         parser.add_argument(
-            "network",
+            "--network",
             metavar="<network>",
             help=_("Name or UUID of the network"))
+        parser.add_argument(
+            '--port',
+            dest='port',
+            metavar='<port>',
+            help=_("Attach to this port (name or UUID).")
+        )
 
         return parser
 
     def take_action(self, parsed_args):
         self.log.debug("take_action(%s)" % parsed_args)
 
         node_uuid = parsed_args.node
-        network_uuid = parsed_args.network
+        if parsed_args.network and parsed_args.port:
+            raise exceptions.CommandError(
+                "ERROR: Specify only one of network or port")
+        if not parsed_args.network and not parsed_args.port:
+            raise exceptions.CommandError(
+                "ERROR: You must specify either network or port")
 
         ironic_client = self.app.client_manager.baremetal
         neutron_client = self.app.client_manager.network
 
+        if parsed_args.network:
+            network = neutron_client.find_network(parsed_args.network)
+            port = None
+        if parsed_args.port:
+            port = neutron_client.find_port(parsed_args.port)
```

Comment:
Just wanted to point out that the second `if` could actually be an `else` since you are already checking that one of  `parsed_args.network` or `parsed_args.port` must exist earlier in the code.
```python
        if parsed_args.network:
            network = neutron_client.find_network(parsed_args.network)
            port = None
        else: # we already checked if parsed_args.network does not exist then parsed_args.port must exist.
            port = neutron_client.find_port(parsed_args.port) 
```
But in the grand scheme of things it probably doesn't matter all that much, so I'll leave it as an optional suggestion. It just saves us one comparison.

Reply:
Good point! I'll update.